source/as-promise/create-rejection.ts

```diff
@@ -0,0 +1,31 @@
+import {CancelableRequest, BeforeErrorHook, RequestError} from './types';
+
+export default function createRejection(error: Error, ...beforeErrorGroups: Array<BeforeErrorHook[] | undefined>): CancelableRequest<never> {
+	const promise = (async () => {
+		if (error instanceof RequestError) {
```

Comment:
Init hooks will throw `RequestError`s if anybody throws inside that hook.

Reply:
`RequestError` provides much more useful information (Got options etc.) that the usual `TypeError` etc.